### ISSUE TITLE
`listingsutf8` with `xelatex` and `lualatex`

### ISSUE THREAD

**Issue by adamsmd** [T-F-S/tcolorbox#56]
The following code compiles fine under PDFLaTeX. However, it fails under XeLaTeX or LuaLaTeX.

```
\documentclass{article}

\usepackage{tcolorbox}
\tcbuselibrary{listingsutf8}

\begin{document}

\begin{tcblisting}{}
$\int x$
\end{tcblisting}

\end{document}
```

As a workaround, I am using `\tcbuselibrary{listings}` instead of `\tcbuselibrary{listingsutf8}`. However, since the documentation for `tcolorbox` recommends using `listingsutf8`, I think it would be a good idea to fix this issue.

Under XeLaTeX, I get the following warning and later on the following error:

```
$ xelatex tcolorbox-listingsutf8-bug.tex

...

Package listingsutf8 Warning: Package loading is aborted because of missing \pdf@unescapehex.
(listingsutf8)                It is provided by pdfTeX >= 1.30.

...

! Package inputenc Error: inputenc is not designed for xetex or luatex.
(inputenc)                only UTF-8 supported.

See the inputenc package documentation for explanation.
Type  H <return>  for immediate help.
 ...igned for xetex or luatex.\MessageBreak only UTF-8 supported.^^J^^JSee the inputenc package documentation for explanation.^^JType  H <return>  for immediate help\@err@

l.10 \end{tcblisting}

?
```

Under LuaLaTeX, I get the following error:

```
$ lualatex tcolorbox-listingsutf8-bug.tex

...

! Package inputenc Error: inputenc is not designed for xetex or luatex.
(inputenc)                only UTF-8 supported.

See the inputenc package documentation for explanation.
Type  H <return>  for immediate help.
 ...ot designed for xetex or luatex.\MessageBreak only UTF-8 supported.

See the inputenc package documentation for explanation.
Type  H <return>  for immediate help\@err@

l.10 \end{tcblisting}

?
```

The version of `tcolorbox` that I have is `2018/01/12 version 4.12 text color boxes`.
I am running this on Ubuntu 18.04.1 LTS.
The versions of PDFLaTeX, XeLaTeX, and LuaLaTeX that I am using are the following.

```
$ pdflatex -version
pdfTeX 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian)
kpathsea version 6.2.3
Copyright 2017 Han The Thanh (pdfTeX) et al.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Han The Thanh (pdfTeX) et al.
Compiled with libpng 1.6.34; using libpng 1.6.34
Compiled with zlib 1.2.11; using zlib 1.2.11
Compiled with poppler version 0.62.0

$ xelatex -version
XeTeX 3.14159265-2.6-0.99998 (TeX Live 2017/Debian)
kpathsea version 6.2.3
Copyright 2017 SIL International, Jonathan Kew and Khaled Hosny.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the XeTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the XeTeX source.
Primary author of XeTeX: Jonathan Kew.
Compiled with ICU version 60.2; using 60.2
Compiled with zlib version 1.2.11; using 1.2.11
Compiled with FreeType2 version 2.8.1; using 2.8.1
Compiled with Graphite2 version 1.3.11; using 1.3.11
Compiled with HarfBuzz version 1.7.2; using 1.7.2
Compiled with libpng version 1.6.34; using 1.6.34
Compiled with poppler version 0.62.0
Compiled with fontconfig version 2.12.6; using 2.12.6

$ lualatex -version
This is LuaTeX, Version 1.0.4 (TeX Live 2017/Debian)

Execute  'luatex --credits'  for credits and version details.

There is NO warranty. Redistribution of this software is covered by
the terms of the GNU General Public License, version 2 or (at your option)
any later version. For more information about these matters, see the file
named COPYING and the LuaTeX source.

LuaTeX is Copyright 2017 Taco Hoekwater and the LuaTeX Team.
```

**Comment by T-F-S** [comment]
This is a misunderstanding, because users of XeLaTeX are explicitly warned to use `listingsutf8`, see Section 15.1.2 on Page 299 or Section 15.4 on Page 308.

`listingsutf8` provides integration of the package `listingsutf8` which maps a *one-byte-encoding* to UTF-8.

So, please, do not use `listingsutf8` with `xelatex` or `lualatex`.

**Comment by adamsmd** [comment]
I agree that this was a misunderstanding on my part. I think the reason for my misunderstanding is that I took the recommendation at the end of the first paragraph of Section 15.1 (page 297 of version 4.12 of the documentation) at face value and didn't realize I needed to read further.

To prevent such misunderstanding in other users as well as to clarify that the existing warnings about XeLaTeX also apply to LuaLaTeX, I would like to suggest something like pull request #58.
